src/tsickle.ts

```diff
@@ -731,25 +771,34 @@ class Annotator extends ClosureRewriter {
       //
       // Both of these forms create a local name "a", which after
       // TypeScript CommonJS compilation will become some renamed
-      // variable like "module_1.a".  But a user might still use plain
-      // "a" in some JSDoc comment, so gather up these local names for
-      // imports and make an alias for each for JSDoc purposes.
+      // variable like "module_1.default" or "module_1.a" (for default vs named bindings,
+      // respectively).
+      // tsickle references types in JSDoc. Because the module prefixes are not predictable, and
+      // because TypeScript might remove imports entirely if they are only for types, the code below
+      // inserts an artificial `const prefix = goog.require` call for the module, and then registers
+      // all symbols from this import to be prefixed.
+      // Subtlety warning: for a default import, the import name is used as the prefix. This means
+      // references to the import do not have to be prefixed, as it already has the correct name.
       if (!this.options.untyped) {
-        let localNames: string[];
-        if (importClause.name) {
+        let bindings = importClause.namedBindings;
+        if (bindings && bindings.kind !== ts.SyntaxKind.NamedImports) {
+          throw new Error(`bindings must be named imports! ${bindings.kind}`);
```

Comment:
Can this happen?  Is this an "I haven't implemented this" or a "notreached()"?

It appears this `bindings` is never used so drop this variable?